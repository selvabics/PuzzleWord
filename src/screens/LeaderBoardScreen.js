/**
 * This is Leader board screen
 * This screen shows list user who played this game and their scores 
 * Try logging in with multiple mail id and play the game
 * So that you can see how the leader board displays the data
 */
import React from 'react';
import { StyleSheet, FlatList, View, Text, Image } from 'react-native';
import { useSelector } from 'react-redux';

import CommonStyles from '../utils/styles';
import { Colors } from '../utils/constants';
import { getUserListSelector } from '../redux/slice/loginSlice';
import { pop } from '../router/navigation';
import ActionButton from '../components/actionButton';
import BackgroundWallpaper from '../components/backgroundWallpaper';
import { responsiveWidth } from 'react-native-responsive-dimensions';
import { Profile } from '../assets';

const LeaderBoardScreen = () => {
    // List of users who logged in to the app
    const userList = [...useSelector(getUserListSelector)].sort((a, b) => (b.totalPoints - a.totalPoints));

    /**
     * Navigate to previous screen
     */
    const goToHomeScreen = () => {
        pop();
    }

    return (
        <BackgroundWallpaper>
            <View style={CommonStyles.flex}>
                <View style={[CommonStyles.center, { margin: 20 }]}>
                    <Text style={CommonStyles.headerText}>Leader Board</Text>
                </View>
                <FlatList
                    contentContainerStyle={CommonStyles.flex}
                    data={userList}
                    renderItem={({ item, index }) => (
                        <View style={styles.rowContainerView}>
                            <View style={[CommonStyles.center, { flexDirection: 'row' }]}>
                                <Image
                                    style={{ width: responsiveWidth(12), height: responsiveWidth(12), borderRadius: responsiveWidth(2) }}
                                    source={Profile}
                                />
                                <View style={{ paddingHorizontal: responsiveWidth(3) }}>
                                    <Text style={CommonStyles.h4}>
                                        {item.email.split('@')[0]}
                                    </Text>
                                    <Text style={CommonStyles.h6}>
                                        {item.email}
                                    </Text>
                                </View>
                            </View>
                            <View style={CommonStyles.center}>
                                <Text style={CommonStyles.h6}>{'Points'}</Text>
                                <Text style={CommonStyles.h3}>{item.totalPoints}</Text>
                            </View>
                        </View>
                    )}
                />
                <View style={styles.footerView}>
                    <ActionButton
                        title={'BACK'}
                        onPress={goToHomeScreen}
                    />
                </View>
            </View>
        </BackgroundWallpaper>
    );
};

const styles = StyleSheet.create({
    rowContainerView: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        margin: responsiveWidth(3),
        marginTop: 0,
        padding: responsiveWidth(2.5),
        backgroundColor: Colors.transparentWhite,
        borderWidth: 1,
        borderRadius: 5,
        borderColor: Colors.secondary,
    },
    footerView: {
        flexDirection: 'row',
        justifyContent: 'center',
        width: '100%',
        padding: 20,
    },
});

export default LeaderBoardScreen;
