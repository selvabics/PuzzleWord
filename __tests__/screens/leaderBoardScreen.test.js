import renderer from 'react-test-renderer';
import LeaderBoardScreen from '../../src/screens/LeaderBoardScreen';

import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';

jest.useFakeTimers();

describe('Testing login screen', () => {
    it('renders Login Screen', () => {
        const mockStore = configureStore()
        const store = mockStore({loginReducer: {userList: []}})
        const tree = renderer
            .create(<Provider store={store}><LeaderBoardScreen /></Provider>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});