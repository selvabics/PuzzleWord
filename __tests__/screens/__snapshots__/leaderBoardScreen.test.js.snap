// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login screen renders Login Screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      {
        "flex": 1,
        "paddingBottom": 20,
        "paddingTop": 30,
      },
      {},
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/background.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "margin": 20,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 45.89389937671454,
            "margin": 15,
            "textAlign": "center",
          }
        }
      >
        Leader Board
      </Text>
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "flex": 1,
        }
      }
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 20,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={false}
          forwardedRef={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(136,213,66,0.8)",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 80.03999999999999,
              "justifyContent": "center",
              "margin": 20,
              "opacity": 0,
              "padding": 7.5,
              "transform": [
                {
                  "scale": 0.3,
                },
              ],
              "width": 525,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 30.595932917809698,
                "fontWeight": "bold",
              }
            }
          >
            BACK
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
